src/UserAdmin/src/Smart.FA.Catalog.Web/Identity/UserIdentity.cs

```diff
@@ -7,26 +7,22 @@ namespace Smart.FA.Catalog.Web.Identity;
 /// <inheritdoc />
 public class UserIdentity : IUserIdentity
 {
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
     /// <inheritdoc />
-    public int Id => Identity.Id;
+    public int Id => Identity?.Id ?? 0;
 
     /// <inheritdoc />
-    public CustomIdentity Identity { get; }
+    public CustomIdentity Identity => (_httpContextAccessor.HttpContext?.User.Identity as CustomIdentity)!;
```

Comment:
It seems we should handled the case where Identity would not be castable to CustomIdentity

Reply:
If we are certain the HttpContext and that Identity cannot be null (since there are no fallback scenarios), then I believe that might be more correct:
```suggestion
    public CustomIdentity Identity => (_httpContextAccessor.HttpContext!.User.Identity as CustomIdentity)!;
```

Reply:
It is already the case : when you seed the data you don't have any identity. But that's a particular case and I don't want to expose to the other developer a nullable api because of one exceptional case.